src/main/java/it/pagopa/pn/deliverypush/action/refinement/RefinementHandler.java

```diff
@@ -41,7 +41,15 @@ public void handleRefinement(String iun, Integer recIndex) {
 
         //Se la notifica è già stata visualizzata non viene perfezionata per decorrenza termini in quanto è già stata perfezionata per presa visione
         if( !isNotificationViewed ) {
-            addRefinementElement(iun, recIndex, pnDeliveryPushConfigs.getRetentionAttachmentDaysAfterRefinement(),true);
+            // recupero la refinementDate dall'element dello schedule refinement
+            Instant refinementDate = timelineUtils.getScheduleRefinement(iun,recIndex).map(scheduleRefinementTimelineElem -> {
```

Comment:
Potremmo prendere anche il notBefore della action, ragioniamoci

Reply:
si e no: ok che oggi sono uguali, ma perchè rischiare di prendere date che magari, in futuro, per motivi XYZ potrebbero cambiare?
Tanto non sarà il costo della getItem su dynamo a far fallire pagopa :D